components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java

```diff
@@ -338,14 +342,19 @@ public static ApplicationAuthenticator getAppAuthenticatorByName(String name) {
     }
 
     /**
-     * @param request
-     * @return
+     * Retrieve authentication context data. This method doesn't check for the expired context.
+     * @deprecated. Please use {{@link #getContextDataWithExpiry(HttpServletRequest)}}.
+     *
+     * @param request Authentication request.
+     * @return Authentication context.
      */
+    @Deprecated
```

Comment:
Instead of deprecating this, can we use the expiry check within this same method based on a config.

Reason for this suggestion is that is that it will be hard to change all the usages to the new method

Reply:
`can we use the expiry check within this same method based on a config.`
We have to use a new method since we are changing the method signature.